Indicators/PivotPointsHighLow.cs

```diff
@@ -0,0 +1,247 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// Pivot Points (High/Low), also known as Bar Count Reversals, indicator.
+    /// https://www.fidelity.com/learning-center/trading-investing/technical-analysis/technical-indicator-guide/pivot-points-high-low
+    /// </summary>
+    public class PivotPointsHighLow : WindowIndicator<IBaseDataBar>
+    {
+        private readonly int _lengthHigh;
+        private readonly int _lengthLow;
+
+        // Indicator will keep information of the last 100 pivot points
+        private readonly RollingWindow<PivotPoint> _pivotPoints = new RollingWindow<PivotPoint>(100);
```

Comment:
This rolling window feels kind of related to feature https://github.com/QuantConnect/Lean/issues/5407, not 100% related to the indicator itself I think. These 100 value count should probably be an argument of `PivotPointsHighLow` I think